ARCs/arc-0084.md

```diff
@@ -0,0 +1,750 @@
+      "args": [
+        {
+          "name": "key",
+          "type": "(uint64,byte[])",
+          "struct": "MetadataKey"
+        },
+        {
+          "name": "data",
+          "type": "byte[]"
+        }
+      ],
+      "returns": {
+        "type": "void"
+      },
+      "actions": {
+        "create": [],
+        "call": [
+          "NoOp"
+        ]
+      }
+    },
+    {
+      "name": "arc84_setAllowance",
+      "args": [
+        {
+          "name": "allowanceKey",
+          "type": "(address,address,uint64)",
+          "struct": "AllowanceKey"
+        },
+        {
+          "name": "allowance",
+          "type": "(uint64,uint64,uint64,uint64,uint64)",
+          "struct": "Allowance"
+        }
+      ],
+      "returns": {
+        "type": "void"
+      },
+      "actions": {
+        "create": [],
+        "call": [
+          "NoOp"
+        ]
+      }
+    },
+    {
+      "name": "arc84_setAllowances",
+      "desc": "********************\nMulti Setter Methods\n********************",
+      "args": [
+        {
+          "name": "allowances",
+          "type": "((address,address,uint64),(uint64,uint64,uint64,uint64,uint64))[]"
+        }
+      ],
+      "returns": {
+        "type": "void"
+      },
+      "actions": {
+        "create": [],
+        "call": [
+          "NoOp"
+        ]
+      }
+    },
+    {
+      "name": "doTransfers",
+      "desc": "*********************\nTransfer/Mint Methods\n*********************",
+      "args": [
+        {
+          "name": "sender",
+          "type": "address"
+        },
+        {
+          "name": "transfers",
+          "type": "(uint64,address,address,uint64)[]"
+        }
+      ],
+      "returns": {
+        "type": "void"
+      },
+      "actions": {
+        "create": [],
+        "call": [
+          "NoOp"
+        ]
+      }
+    },
+    {
+      "name": "arc84_mint",
+      "args": [
+        {
+          "name": "collectionId",
+          "type": "uint64"
+        },
+        {
+          "name": "params",
+          "type": "(byte[32],byte[8],uint64,uint64,address,uint64)",
+          "struct": "Params"
+        }
+      ],
+      "returns": {
+        "type": "uint64"
+      },
+      "actions": {
+        "create": [],
+        "call": [
+          "NoOp"
+        ]
+      }
+    }
+  ]
+}
+```
+
+</details>
+
+#### TEALScript Data Interface Signatures
+
+```typescript
+
+export type Params = {
+  /** The descriptive name of the token */
+  name: bytes<32>;
+  /** The short ticker symbol of the token */
+  symbol: bytes<8>;
+  /** The total supply */
+  total: uint64;
+  /** The number of digits to use after the decimal point when displaying the asset */
+  decimals: uint64;
+  /** The address that can modify the token metadata */
+  manager: Address;
+  /** The address of the app that implements the transfer hook. If this is not set, transfers are always approved */
+  transferHookApp: AppID;
+};
```

Comment:
You're missing the url field from the asset params.

Reply:
I originally intentionally omitted it because I thought it could just be a metadata field, but since it's such an essential param for clients probably worth it to add to the params

Reply:
You can't say it's an ASA upgrade/replacement and remove the literal one field many need for token and nft images (and that want it to be mutable).  😀

Reply:
Yeah that's fair. I was originally thinking we could just define a standard for how the URL is specified in the metadata, but adding it to params just makes way more sense for ASA compatibility.